mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/InVmEventDelivery.java

```diff
@@ -72,19 +73,29 @@ public ExecutionStages deliver(Collection<MailboxListener> mailboxListeners, Eve
 
     private Mono<Void> doDeliver(Collection<MailboxListener> mailboxListeners, Event event) {
         return Flux.fromIterable(mailboxListeners)
-            .flatMap(mailboxListener -> Mono.fromRunnable(() -> doDeliverToListener(mailboxListener, event)))
+            .flatMap(mailboxListener -> deliveryWithRetries(event, mailboxListener))
             .then()
             .subscribeOn(Schedulers.elastic());
     }
 
+    private Mono<Object> deliveryWithRetries(Event event, MailboxListener mailboxListener) {
+        return Mono.fromRunnable(() -> doDeliverToListener(mailboxListener, event))
+            .retry(MAX_RETRIES)
```

Comment:
There is as far as I know no exponential backoff delays here. The idea:

 - execute directly
 - If it fails wait 100 ms (eg) then retry
 - If it fails wait 200 ms then retry
 - If it fails wait 400 ms then retry
 - If it fails wait 800 ms then retry

I do not see such delays here

Reply:
I did not implement it

Reply:
... yet ^^